quality_control_stock_oca/models/qc_inspection.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2014 Serv. Tec. Avanzados - Pedro M. Baeza
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.fields import first
+
+
+class QcInspection(models.Model):
+    _inherit = "qc.inspection"
+
+    picking_id = fields.Many2one(
+        comodel_name="stock.picking", compute="_compute_picking", store=True
+    )
+    lot_id = fields.Many2one(
+        comodel_name="stock.lot", compute="_compute_lot", store=True
+    )
+
+    def object_selection_values(self):
+        result = super().object_selection_values()
+        result.extend(
+            [
+                ("stock.picking", "Picking List"),
+                ("stock.move", "Stock Move"),
+                ("stock.lot", "Lot/Serial Number"),
+            ]
+        )
+        return result
+
+    @api.depends("object_id")
+    def _compute_picking(self):
+        for inspection in self:
+            if inspection.object_id:
+                if inspection.object_id._name == "stock.move":
+                    inspection.picking_id = inspection.object_id.picking_id
+                elif inspection.object_id._name == "stock.picking":
+                    inspection.picking_id = inspection.object_id
+
+    @api.depends("object_id")
+    def _compute_lot(self):
+        moves = self.filtered(
+            lambda i: i.object_id and i.object_id._name == "stock.move"
+        ).mapped("object_id")
+        move_lines = self.env["stock.move.line"].search(
+            [("lot_id", "!=", False), ("move_id", "in", [move.id for move in moves])]
+        )
+
+        for inspection in self:
+            if inspection.object_id:
+                if inspection.object_id._name == "stock.move.line":
```

Comment:
Still this `stock.move.line` unnecessary reference. This is the last one IMO

Reply:
Done